pkg/registry/core/pod/strategy.go

```diff
@@ -92,6 +94,14 @@ func (podStrategy) AllowCreateOnUpdate() bool {
 // ValidateUpdate is the default update validation for an end user.
 func (podStrategy) ValidateUpdate(ctx genericapirequest.Context, obj, old runtime.Object) field.ErrorList {
 	errorList := validation.ValidatePod(obj.(*api.Pod))
```

Comment:
Why shouldn't a pod be allowed to be invalid during the process of initialization?  It is possible to do with admission today and given a "my special defaulter that can take an invalid user object and choose a good/valid default" initializer, seems even more useful after.

Reply:
not in a persisted, visible state that someone getting an object from the API has to deal with.

Reply:
> not in a persisted, visible state that someone getting an object from the API has to deal with, or that migration has to take into account. bad data in etcd is not something I'm excited about.

It's uninitialized data.  Not allowing someone to initialize that data to a shape of their choosing seems excessively restrictive.  I know that I would respond to an API not letting me do it by pushing garbage into fields that fail validation and using magic values.

Our API already has problems dealing with new fields and I don't see forcing this limitation as helping.

Reply:
if we want to further relax this in the future, I think we can, if there are compelling use cases that outweigh the downsides of persisting bad data.

Reply:
> if we want to further relax this in the future, I think we can, if there are compelling use cases that outweigh the downsides of persisting bad data.

I suppose its true that we can't go the other way around.  If we do it, I reserve the right to say "I told you so" twice. :)

Reply:
> seems even more useful after.

It might be useful. However, user can GET uninitialized objects. I think it's more important to never persist invalid objects.

Reply:
> I suppose its true that we can't go the other way around. If we do it, I reserve the right to say "I told you so" twice. :)

let the record show that @deads2k and @liggitt do not always agree :)